packages/app/src/components/CardInfoLifecycle.tsx

```diff
@@ -1,31 +1,192 @@
+    description: 'Card not usable anymore',
+    sheetDescription:
+      'The issuer has revoked this card and it can not be used anymore. Contact the issuer for more information.',
+  },
+  // We can hardcode this to the rules for the PID credential as this will be the only of this type for now.
+  batch: {
+    variant: 'warning',
+    title: 'Limited card usage',
+    description: 'verifications left',
+    sheetDescription:
+      'This card requires periodic validation using an internet connection. When usage is low you will be notified.',
+  },
+}
+
+interface CardInfoLifecycleProps {
+  validUntil?: Date
+  validFrom?: Date
+  isRevoked?: boolean
+  batchLeft?: number
+}
+
+export function CardInfoLifecycle({ validUntil, validFrom, isRevoked, batchLeft }: CardInfoLifecycleProps) {
+  const toast = useToastController()
+  const [isOpen, setIsOpen] = useState(false)
+  const { withHaptics } = useHaptics()
+
+  const state = useMemo(() => {
+    if (isRevoked) return 'revoked'
+    if (batchLeft) return 'batch'
+
+    return 'active'
+  }, [isRevoked, batchLeft])
+
+  const onPress = withHaptics(() => setIsOpen(!isOpen))
+
+  const onPressValidate = withHaptics(() => {
+    // Implement navigation to the setup eID card flow.
+    toast.show('Coming soon', { customData: { preset: 'warning' } })
+  })
+
+  if (validUntil || validFrom) {
+    return <CardInfoLimitedByDate validUntil={validUntil} validFrom={validFrom} />
+  }
+
+  return (
+    <>
+      {batchLeft && batchLeft <= 5 && (
+        <Stack pb="$4">
+          <Button.Solid bg="$grey-50" bw="$0.5" borderColor="$grey-200" color="$grey-900" onPress={onPressValidate}>
+            Refresh card <HeroIcons.ArrowPath ml="$-2" size={20} color="$grey-700" />
+          </Button.Solid>
+        </Stack>
+      )}
+      <InfoButton
+        routingType="modal"
+        variant={cardInfoLifecycleVariant[state].variant}
+        title={cardInfoLifecycleVariant[state].title}
+        description={
+          batchLeft
+            ? `${batchLeft} ${cardInfoLifecycleVariant[state].description}`
+            : cardInfoLifecycleVariant[state].description
+        }
+        onPress={onPress}
+      />
+      <InfoSheet
+        isOpen={isOpen}
+        setIsOpen={setIsOpen}
+        onClose={onPress}
+        variant={cardInfoLifecycleVariant[state].variant}
+        title={cardInfoLifecycleVariant[state].title}
+        description={cardInfoLifecycleVariant[state].sheetDescription}
+      />
+    </>
+  )
+}
+
+type CardInfoLimitedByDateState = 'not-yet-active' | 'active' | 'will-expire' | 'expired'
+
+function CardInfoLimitedByDate({ validUntil, validFrom }: { validUntil?: Date; validFrom?: Date }) {
+  const [state, setState] = useState<CardInfoLimitedByDateState>('active')
   const [isOpen, setIsOpen] = useState(false)
   const { withHaptics } = useHaptics()
 
   const onPress = withHaptics(() => setIsOpen(!isOpen))
 
+  useEffect(() => {
+    if (validFrom && validFrom > new Date()) {
```

Comment:
Can you compare dates like that? Don't you have to call getTime?

Reply:
Also i'd handle expired before validFrom. If somehow it's validUntil is before validFrom

Reply:
Yes it works like this.

I've moved the expired 👍